### ISSUE TITLE
Tenants client `create_or_replace` requires default_preferences when it should not

### ISSUE THREAD

**Issue by ryancausey** [trycourier/courier-python#71]
The behavior of `create_or_replace` for the Tenants client does not match the REST API nor its Python method signature.

When I attempt to create a tenant without specifying `default_preferences`, I get the following error:

```python
from courier import Client
client = Courier(authorization_token="******")
client.tenants.create_or_replace(tenant_id="org_asdf", name="foobaz")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/rccausey/.cache/pypoetry/virtualenvs/notifications-api-6mUc-8uv-py3.11/lib/python3.11/site-packages/courier/tenants/client.py", line 115, in create_or_replace
    return pydantic_v1.parse_obj_as(Tenant, _response.json())  # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rccausey/.cache/pypoetry/virtualenvs/notifications-api-6mUc-8uv-py3.11/lib/python3.11/site-packages/pydantic/v1/tools.py", line 38, in parse_obj_as
    return model_type(__root__=obj).__root__
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/rccausey/.cache/pypoetry/virtualenvs/notifications-api-6mUc-8uv-py3.11/lib/python3.11/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for ParsingModel[Tenant]
__root__ -> default_preferences -> items
  field required (type=value_error.missing)
```

This same call to the REST API directly works fine.

The API documentation here states that `default_preferences` are not required, which also matches the Python method signature: https://www.courier.com/docs/reference/tenants/create-replace/

This was observed with trycourier = "6.0.2".

**Comment by dsinghvi** [comment]
@ryancausey thanks for filing this issue -- we'll diagnose and get back to you shortly.

**Comment by dsinghvi** [comment]
@ryancausey this is fixed in [v6.0.3](https://github.com/trycourier/courier-python/releases/tag/v6.0.3)!
